docs/tutorial/sandbox.md

```diff
@@ -0,0 +1,136 @@
+# Process Sandboxing
+
+One key security feature in Chromium is that processes can executed within a sandbox.
+The sandbox limits the harm that untrusted code can cause by limiting access to most
+system resources — sandboxed processes can only freely use CPU cycles and memory.
+In order to perform operations requiring additional privilege, sandboxed processes
+use dedicated communication channels to delegate tasks to more privileged processes.
+
+In Chromium, sandboxing is mainly applied to renderer processes. In addition, certain
+utility processes are also sandboxed, including the audio service, the GPU service,
+and the storage service. On Windows, the `chrome://sandbox` page will show you the exact
+sandboxing policy associated with each process.
+
+See Chromium's [Sandbox design document][sandbox] for more information.
+
+## Sandboxing in Electron
+
+Electron comes with a mixed sandbox environment, having Chromium's sandbox enabled but
+disabling it by default for renderer processes (meaning that it still applies to the
+relevant utility processes).
+
+Unsandboxed renderers are not a problem for desktop applications that only display
+trusted code, but it makes Electron less secure than Chromium for displaying untrusted
+web content. For applications that require more security, sandboxed renderers are
+recommended.
+
+Sandboxed processes in Electron behave _mostly_ in the same way as Chromium's do, but
+Electron has a few additional quirks to consider.
+
+### Renderer processes
+
+When renderer processes in Electron are sandboxed, they behave in the same way as a
+regular Chrome renderer would. A sandboxed renderer won't be initialized in a Node.js
+environment, nor will it modify any default JavaScript APIs that behave differently
+by default in Electron (e.g. `window.open`).
+
+<!-- TODO(erickzhao): when we have a solid guide for IPC, link it here -->
+Therefore, when the sandbox is enabled, renderer processes can only make changes to the
+system by delegating tasks to the main process via inter-process communication (IPC).
+
+### Preload scripts
+
+In order to allow renderer processes to communicate with the main process, preload
+scripts attached to sandboxed renderers will still have a limited Node.js environment
+enabled. Node's `require` module is exposed, but will only be able to import a subset
+of Electron and Node's built-in modules:
```

Comment:
```suggestion
scripts attached to sandboxed renderers will still have a polyfilled subset of Node.js APIs
available. A `require` function similar to Node's `require` module is exposed, but can only
import a subset of Electron and Node's built-in modules:
```

We might want to reference somewhere in this section that because the `require` function only provides access to these four modules users writing preload scripts may have to bundle their preload script into a single file utilizing bundlers such as webpack or parcel.

Reply:
Oh, the previous iteration of this doc had that mentioned (although it used browserify as an example).

 Could you expand on the importance of bundling the preload script into a single file? I don't think I follow how that relates to the polyfilled `require` function. Is it because they may want to use `require` with other modules?

Reply:
> Could you expand on the importance of bundling the preload script into a single file?

This require function is less of a `require` function and more of a `getModule()` function that can only return four things.  In normal Node.js code people will split their code across multiple files and use `require('./thing')` to include the secondary file from the primary file.  This is not possible in sandbox mode as there is no disk access from the sandboxed process and our require function was never built to support that.  Therefore anyone writing their preload script in a moduluar way will need to bundle it into a single file using a pre-processor like webpack or browserify.

Reply:
Thank you! I'll make sure to make that clear.

Reply:
Addressed in c0870e7